pkg/types/vsphere/validation/platform.go

```diff
@@ -29,8 +29,10 @@ func ValidatePlatform(p *vsphere.Platform, fldPath *field.Path) field.ErrorList
 		allErrs = append(allErrs, field.Required(fldPath.Child("defaultDatastore"), "must specify the default datastore"))
 	}
 
-	if strings.ToLower(p.VCenter) != p.VCenter {
-		allErrs = append(allErrs, field.Invalid(fldPath.Child("vCenter"), p.VCenter, "must be all lower case"))
+	if len(p.VCenter) != 0 {
+		if err := validate.VCenter(p.VCenter); err != nil {
+			allErrs = append(allErrs, field.Invalid(fldPath.Child("vCenter"), p.VCenter, "must be the hostname of the vCenter"))
```

Comment:
After reading a bit more on the RFCs, the term hostname is not quite inclusive enough here.
```suggestion
			allErrs = append(allErrs, field.Invalid(fldPath.Child("vCenter"), p.VCenter, "must be the hostname or IP address of the vCenter"))
```

Although in the `vCenter` field description, we use domain name instead of hostname. We should probably stay consistent.
```suggestion
			allErrs = append(allErrs, field.Invalid(fldPath.Child("vCenter"), p.VCenter, "must be the domain name or IP address of the vCenter"))
```